src/Operation/ChangeStream.php

```diff
@@ -0,0 +1,159 @@
+use MongoDB\Exception\InvalidArgumentException;
+use MongoDB\Exception\UnexpectedValueException;
+use MongoDB\Exception\UnsupportedException;
+use stdClass;
+use Traversable;
+
+/**
+ * Operation for the changeStream command.
+ *
+ * @api
+ * @see \MongoDB\Collection::changeStream()
+ * @see http://docs.mongodb.org/manual/reference/command/changeStream/
+ */
+class ChangeStream implements Executable
+{
+    private $databaseName;
+    private $collectionName;
+    private $resumeToken;
+    private $pipeline;
+    private $readPreference;
+    private $options;
+
+    /**
+     * Constructs a changeStream command.
+     *
+     * Supported options:
+     *
+     *  * fullDocument (string): Allowed values: ‘default’, ‘updateLookup’.
+     *    Defaults to ‘default’.  When set to ‘updateLookup’, the change
+     *    notification for partial updates will include both a delta describing
+     *    the changes to the document, as well as a copy of the entire document
+     *    that was changed from some time after the change occurred. For forward
+     *    compatibility, a driver MUST NOT raise an error when a user provides
+     *    an unknown value. The driver relies on the server to validate this
+     *    option.
+     *
+     *  * resumeAfter (document): Specifies the logical starting point for the
+     *    new change stream.
+     *
+     *  * maxAwaitTimeMS (integer): The maximum amount of time for the server to
+     *    wait on new documents to satisfy a change stream query.
+     *
+     *  * batchSize (integer): The number of documents to return per batch.
+     *
+     *    This option is sent only if the caller explicitly provides a value.
+     *    The default is to not send a value.
+     *
+     *  * collation (document): Specifies a collation.
+     *
+     *    This option is sent only if the caller explicitly provides a value.
+     *    The default is to not send a value.
+     *
+     * @param string         $databaseName   Database name
+     * @param string         $collectionName Collection name
+     * @param array          $pipeline       List of pipeline operations
+     * @param array          $options        Command options
+     * @throws InvalidArgumentException for parameter/option parsing errors
+     */
+    public function __construct($databaseName, $collectionName, array $pipeline, array $options = [])
+    {
+        $expectedIndex = 0;
+
+        foreach ($pipeline as $i => $operation) {
+            if ($i !== $expectedIndex) {
+                throw new InvalidArgumentException(sprintf('$pipeline is not a list (unexpected index: "%s")', $i));
+            }
+
+            if ( ! is_array($operation) && ! is_object($operation)) {
+                throw InvalidArgumentException::invalidType(sprintf('$pipeline[%d]', $i), $operation, 'array or object');
+            }
+
+            $expectedIndex += 1;
+        }
+
+        if (isset($options['fullDocument']) && ! is_string($options['fullDocument'])) {
```

Comment:
Let's create `DEFAULT` and `UPDATE_LOOKUP` constants for now. Per the spec, we don't need to enforce that the user provides one of those two values.